src/option.h

```diff
@@ -452,10 +452,12 @@ namespace librealsense
     class polling_errors_disable : public option
     {
     public:
-        polling_errors_disable(polling_error_handler* handler)
-            : _polling_error_handler(handler), _value(1)
+        polling_errors_disable(std::unique_ptr<polling_error_handler>& handler)
+            : _polling_error_handler(std::move(handler)), _value(1)
```

Comment:
Not sure I like this: why not make it a shared_ptr<> instead, then we don't run into a situation where the value is now null but there really still *is* a handler?

Reply:
The fix addresses only the issue of the polling handler in 'orphan' mode after device is (forcibly) disconnected.
In that case the device is indeed is 0xdeadbeef but the handler remained active and could fail on dereferencing invalid pointer.
With this design the handler will remain intact, and after device disconnects it will stop the active object in dtor.
So the polling events will be handled properly during the period between device disconnect and the dtor call

Reply:
I also recognize more things to improve in the design of active objects lifetime control but this will require a separate session